src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +34,13 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        String[] trimmedAddressParts = trimmedAddress.split(", ");
+        blockNumber = new Block(trimmedAddressParts[BLOCK_NUMBER].replace("a/", "")); //Remove the '/a' from block number
```

Comment:
`String#replace()` replaces all occurrences regardless of position. Consider using `String#substring()` on `trimmedAddressParts` before splitting it up instead, as that is more intuitive.